cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/class-media.php

```diff
@@ -146,19 +180,71 @@ public function is_media( $attachment_id ) {
 	}
 
 	/**
-	 * Get a resource type based on file.(Cloudinary v1 remove mime type in post data).
+	 * Convert media extension.
 	 *
-	 * @param \WP_Post|int $attachment_id The attachment ID or object.
+	 * @param string $filename The file to convert.
+	 *
+	 * @return string|null
+	 */
+	public function convert_media_extension( $filename ) {
+
+		// Add preferred formats in future.
+		$conversion_types = array(
+			'psd' => 'jpg',
```

Comment:
@DavidCramer I thought I had to add possible conversions to this array?

Reply:
yes, sorry, I misunderstood your comment 🤦 
It needs (and testing) all the adobe formats.
Take a look at https://cloudinary.com/documentation/image_transformations#supported_image_formats 
You can then find the formats that are not supported by WordPress, and add them in.

Reply:
Thanks for the link! I added them to the array.